client/rpc_sender.go

```diff
@@ -83,6 +84,13 @@ func newRPCSender(server string, context *base.Context, retryOpts retry.Options,
 // and been executed successfully. We retry here to eventually get through with
 // the same client command ID and be given the cached response.
 func (s *rpcSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.Error) {
+	timeout := 5 * s.retryOpts.MaxBackoff
```

Comment:
> All that said, I think this patch is the best we can hope for until go 1.6.

This patch effectively changes the connect timeout from 250ms to 25s, which seems excessive. It would be more appropriate to change the `InitialBackoff` in use. 

Go 1.6 will help a bit, but it's only a factor of 2 for RSA operations (compared to the 20x improvement for AES-GCM).